src/engine/irc.cpp

```diff
@@ -357,19 +357,19 @@ ICOMMAND(0, ircbind, "ss", (const char *name, const char *s), {
 ICOMMAND(0, ircpass, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current password is: %s", n->passkey && *n->passkey ? "<set>" : "<not set>"); return; }
+    if(!s || !*s) { ircprintf(n, 4, NULL, "current password is: %s", *n->passkey ? "<set>" : "<not set>"); return; }
```

Comment:
Because the address of an array always returns true when compared against something? Even if the array is empty, that's what it will give.

Consider the following toy code:
```c
#include <stdio.h>
#include <stdbool.h>

int main() {
	int hello[5];
	bool aa = true;

	if (hello && aa)
		puts("I'm true!");
	else
		puts("I'm not!");

	printf("%d\n", hello);

	return 0;
}
```
Even if you aggregate-initialize that `hello[5]`, it will always be true and the compiler rightfully says so as well.